Makefile

```diff
@@ -1,6 +1,39 @@
+.PHONY: make-artifacts-dir generate validate-sigs test coverage install-metrics-binaries lint
+
+export GOLANGCI_LINT_VERSION := v1.62.2
+ifndef GOPATH
+    GOPATH=$(shell go env GOPATH)
+    export GOPATH
+endif
+ifndef ARTIFACTS
+	ARTIFACTS=/tmp/artifacts
+    export ARTIFACTS
+endif
+ifndef COVERAGE_OUTPUT_PATH
+	COVERAGE_OUTPUT_PATH=${ARTIFACTS}/coverage.html
+    export COVERAGE_OUTPUT_PATH
+endif
+
+make-artifacts-dir:
+	mkdir -p ${ARTIFACTS}
+
 generate:
 	go run ./validators/cmd/sigs --dry-run=false
 	go run ./generators/cmd/sigs
 
 validate-sigs:
 	go run ./validators/cmd/sigs
+
+test:
+	go test ./...
+
+coverage: make-artifacts-dir
+	if ! command -V covreport; then go install github.com/cancue/covreport@latest; fi
+	go test ./... -coverprofile=/tmp/coverage.out
+	covreport  -i /tmp/coverage.out -o ${COVERAGE_OUTPUT_PATH}
+
+install-metrics-binaries:
```

Comment:
nit - Any reason to have another target with an unrelated name for installing a linting tool?

Reply:
probably over-engineering :)